### PR TITLE
backport: Merge bitcoin#22408 ,20772, 21056, 22257, 19776, 20599

### PR DESCRIPTION
**Pull request by vijaydasmp**
This PR and does not introduce breaking changes

## Issue being fixed or feature implemented
Bitcoin backports

## What was done?
backports bitcoin PRS 22408 ,20772, 21056, 22257, 19776, 20599

## How Has This Been Tested?
In CI

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by PastaPastaPasta — APPROVED** [review]
utACK cfa51c3c22ecc7e02e1f2269576bc010c6d49773

File: doc/release-notes-21056.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+New bitcoin-cli settings
+------------------------
+
+- A new `-rpcwaittimeout` argument to `bitcoin-cli` sets the timeout
+  in seconds to use with `-rpcwait`. If the timeout expires,
+  `bitcoin-cli` will report a failure. (#21056)
```
Comment (UdjinM6) [inline doc/release-notes-21056.md:6]: 21056: dashify
```suggestion
New dash-cli settings
------------------------

- A new `-rpcwaittimeout` argument to `dash-cli` sets the timeout
  in seconds to use with `-rpcwait`. If the timeout expires,
  `dash-cli` will report a failure. (dash#5923)
```

**Review by UdjinM6 — DISMISSED** [review]

**Comment by PastaPastaPasta** [comment]
I committed Udjin's suggestion; and then squashed it into the original backport commit.

**Review by PastaPastaPasta — APPROVED** [review]
self-utACK 5dcb666d217e88bdb8e0924dda8bae845b6cf966

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**